stratum/hal/lib/common/hal.cc

```diff
@@ -312,7 +312,11 @@ ::util::Status Hal::RegisterSignalHandlers() {
   // Register the signal handlers and save the old handlers as well.
   std::vector<int> sig = {SIGINT, SIGTERM, SIGUSR2};
   for (const int s : sig) {
-    sighandler_t h = signal(s, SignalRcvCallback);
+    #ifdef __APPLE__
+    	sig_t h = signal(s, SignalRcvCallback);
+    #else
```

Comment:
Not a huge fan of having platform specific code inline with implementations. This could probably be a `typedef`.

Reply:
In fact, it's the same type:


```c
typedef void (*sig_t) (int);
```

https://developer.apple.com/library/archive/documentation/System/Conceptual/ManPages_iPhoneOS/man3/signal.3.html


```c
typedef void (*sighandler_t)(int);
```

https://man7.org/linux/man-pages/man2/signal.2.html